source/lib/src/prod_env_mat.cc

```diff
@@ -175,6 +175,111 @@ prod_env_mat_r_cpu(
   }
 }
 
+template<typename FPTYPE>
+void
+deepmd::
+prod_env_mat_a_mix_cpu(
```

Comment:
We should be able to implement it within `prod_env_mat_a_cpu`. 

Adding one parameter `const int * f_type` with default value `type` to `prod_env_mat_a_cpu` would work.

Reply:
The same comment for the cuda and rocm variances  of `prod_env_mat`